source/Symbol/SwiftASTContext.cpp

```diff
@@ -6905,6 +6905,25 @@ CompilerType SwiftASTContext::GetChildCompilerTypeAtIndex(
   } break;
 
   case swift::TypeKind::Tuple: {
+    // Dynamic type resolution may actually change(!) the layout of a Tuple, so
+    // we need to get the offset from the static (but archetype-bound) version.
+    auto static_value = valobj->GetStaticValue();
+    auto static_type = static_value->GetCompilerType();
+    auto static_swift_type = GetCanonicalSwiftType(static_type);
+    if (static_swift_type->getKind() == swift::TypeKind::Tuple)
```

Comment:
Instead can you write `if (swift::isa<TupleType>(static_swift_type))`?

Reply:
Sure.. why is that preferred, because it's shorter?

Reply:
To answer my own question: I guess it would also work for subclasses of TupleType.